.github/workflows/test-telemetry-setup.yaml

```diff
@@ -1,51 +1,89 @@
+      id: export
+      run: echo service_name="${OTEL_SERVICE_NAME}" >> ${GITHUB_OUTPUT}
+  child-workflow:
+    needs: base-env-setup
+    secrets: inherit
+    uses: rapidsai/shared-actions/.github/workflows/test-child-workflow.yaml@telemetry-dispatch-actions
+  summarize-top-level:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        version: [1, 2]
-        os: [ubuntu-latest] # , windows-latest
+    continue-on-error: true
+    needs:
+        - base-env-setup
+        - child-workflow
     steps:
-      - name: Checkout actions
-        uses: actions/checkout@v4
+      - name: Load base env vars, including OTEL_SERVICE_NAME
+        uses: rapidsai/shared-actions/telemetry-dispatch-load-base-env-vars@telemetry-dispatch-actions
         with:
-            ref: ${{env.SHARED_ACTIONS_REF}}
-            path: ./shared-actions
-      # Run job with traceparent. We'll validate that this matches in the summary data.
-      - name: Get job traceparent in matrix job
-        uses: ./shared-actions/telemetry-traceparent
-        id: job-traceparent
-      - name: Generate traceparent for a step
-        uses: ./shared-actions/telemetry-traceparent
-        id: step-traceparent
+            load_service_name: "true"
+      - name: Check if service name took on an unexpected value
+        run: |
+          echo "(should be the value set to the OTEL_SERVICE_NAME env var in base-env-setup job)"
+          [ "${OTEL_SERVICE_NAME}" = "${{needs.base-env-setup.outputs.service-name}}" ] || exit 1
+      - name: Telemetry summarize
+        uses: rapidsai/shared-actions/telemetry-dispatch-write-summary@telemetry-dispatch-actions
         with:
-            step_name: "Download gha-tools with git clone"
-      - name: Echo computed step traceparent
+          cert_concat: "${{ secrets.OTEL_EXPORTER_OTLP_CA_CERTIFICATE }};${{ secrets.OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE }};${{ secrets.OTEL_EXPORTER_OTLP_CLIENT_KEY }}"
+      - name: Check if service name was altered during telemetry summary
         run: |
-            echo "GHA tools clone job traceparent: ${{ steps.step-traceparent.outputs.traceparent }}"
-      - name: Test OTel export of job JSON
-        uses: ./shared-actions/telemetry-summarize
+          echo "(should be the value set to the OTEL_SERVICE_NAME env var in base-env-setup job)"
+          [ "${OTEL_SERVICE_NAME}" = "${{needs.base-env-setup.outputs.service-name}}" ] || exit 1
+      - name: Query the Tempo HTTP API and check that our trace is present and has expected properties
+        run: |
+          TRACE_ID=$( cut -d '-' -f 2 <<< "$TRACEPARENT" );
+          echo "Trace ID is: ${TRACE_ID}";
+          TRACE_URL="${OTEL_EXPORTER_OTLP_ENDPOINT/4318/3200}/api/traces/${TRACE_ID}"
+          echo "Trace URL is: ${TRACE_URL}"
+          curl \
+            --cert /tmp/certs/client.crt.pem --key /tmp/certs/client.key.pem --cacert /tmp/certs/ca.crt.pem \
+            -Gs "${TRACE_URL}" > trace_record.json;
```

Comment:
just an FYI, this won't work when we switch the `OTEL_EXPORTER_OTLP_ENDPOINT` var to Fluent Bit (since Fluent Bit doesn't store the traces & since we'll be removing the certs from the workflows which will prevent direct Tempo access)

Reply:
Good point. Maybe Grafana Cloud free tier can be a substitute here: https://grafana.com/docs/tempo/latest/getting-started/

Reply:
hmm. i'd definitely like to avoid creating an account unless it's absolutely critical. can we just skip testing the `read` step?

Reply:
Of course we can skip the tests, but then they don't help us. We can also just tell people to run a local tempo instance and do port forwarding to run it on their own machines.

Reply:
In practice I think most actions are tested downstream (in RAPIDS library repositories) anyway. I think it's fine to avoid complication here even if we don't have good tests in this repo.